### PR TITLE
feat: Add intervalprocessor component

### PR DESCRIPTION
**Pull request by ArthurSens**
#### PR Description

Adds [intervalprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/intervalprocessor) as a new component to alloy.

#### PR Checklist

- [X] CHANGELOG.md updated
- [X] Documentation added
- [ ] Tests updated
- [ ] Config converters updated

### CODE REVIEW

**Comment by ArthurSens** [comment]
It's my very first time reading alloy's codebase, so bear with me if I missed anything 😅.

I'm seeing errors when testing this locally, but I'm struggling a bit to understand why

**Comment by clayton-cornell** [comment]
@grafana/grafana-alloy-maintainers I'll circle back to do a doc review once the code base is reviewed.

File: docs/sources/reference/components/otelcol/otelcol.processor.interval.md
Comment (ptodev) [inline docs/sources/reference/components/otelcol/otelcol.processor.interval.md:53]:
```suggestion
`interval`    | `duration` | The interval in which the processor should export the aggregated metrics. | `"60s"` | no
```

File: docs/sources/reference/components/otelcol/otelcol.processor.interval.md
Comment (ptodev) [inline docs/sources/reference/components/otelcol/otelcol.processor.interval.md:109]:
```suggestion
otelcol.processor.interval "default" {
  interval = "30s"
```

File: docs/sources/reference/components/otelcol/otelcol.processor.interval.md
Comment (ptodev) [inline docs/sources/reference/components/otelcol/otelcol.processor.interval.md:111]:
```suggestion
    metrics = [otelcol.exporter.otlphttp.grafana_cloud.input]
```

File: docs/sources/reference/components/otelcol/otelcol.processor.interval.md
Comment (ptodev) [inline docs/sources/reference/components/otelcol/otelcol.processor.interval.md:119]:
```suggestion
otelcol.exporter.otlphttp "grafana_cloud" {
  client {
    endpoint = "https://otlp-gateway-prod-gb-south-0.grafana.net/otlp"
    auth     = otelcol.auth.basic.grafana_cloud.handler
  }
}

otelcol.auth.basic "grafana_cloud" {
  username = env("GRAFANA_CLOUD_USERNAME")
  password = env("GRAFANA_CLOUD_API_KEY")
}
```

It is not possible to send OTLP signals to `prometheus.remote_write` directly. The metrics need to go through `otelcol.exporter.prometheus` first, which will convert them from OTLP format to Prometheus format.

File: docs/sources/reference/components/otelcol/otelcol.processor.interval.md
Comment (ptodev) [inline docs/sources/reference/components/otelcol/otelcol.processor.interval.md:122]:
```suggestion
```

I think this should be part of the previous section. It can explain how the example works, in more detail.

File: docs/sources/reference/components/otelcol/otelcol.processor.interval.md
Comment (ptodev) [inline docs/sources/reference/components/otelcol/otelcol.processor.interval.md:95]:
```suggestion
## Example
```

File: CHANGELOG.md
Comment (ptodev) [inline CHANGELOG.md:16]:
```suggestion
- (_Experimental_) Add an `otelcol.processor.interval` component to aggregate metrics and periodically
  forward the latest values to the next component in the pipeline.
```

**Comment by ArthurSens** [comment]
I do not understand the test failure, I haven't changed anything on the test or on the files used in this test 🤔

**Comment by mattdurham** [comment]
The OSX tests are always flaky and dont stop a merge. Will see what drone reports on the linux tests.

File: go.mod
Comment (mattdurham) [inline go.mod:424]: The error is from this line, how the mapstructure reports errors has changed. You need to update the verbiage in the corrupt_config.diags to match the new error it reports.

Comment (ArthurSens) [reply go.mod:424]: Nice find, thanks!

**Review by mattdurham — APPROVED** [review]
Looks reasonable to me, will wait on Claytons doc review

File: docs/sources/reference/components/otelcol/otelcol.processor.interval.md
Comment (clayton-cornell) [inline docs/sources/reference/components/otelcol/otelcol.processor.interval.md:5]:
```suggestion
description: Learn about otelcol.processor.interval
```

We don't need aliases set on new topics. This should only be used if we move or rename a topic.

File: docs/sources/reference/components/otelcol/otelcol.processor.interval.md
Comment (clayton-cornell) [inline docs/sources/reference/components/otelcol/otelcol.processor.interval.md:15]:
```suggestion
`otelcol.processor.interval`  aggregates metrics and periodically forwards the latest values to the next component in the pipeline.
The processor supports aggregating the following metric types:
```

File: docs/sources/reference/components/otelcol/otelcol.processor.interval.md
Comment (clayton-cornell) [inline docs/sources/reference/components/otelcol/otelcol.processor.interval.md:21]:
```suggestion
The following metric types will *not* be aggregated and will instead be passed, unchanged, to the next component in the pipeline:
```

File: docs/sources/reference/components/otelcol/otelcol.processor.interval.md
Comment (clayton-cornell) [inline docs/sources/reference/components/otelcol/otelcol.processor.interval.md:29]:
```suggestion
After exporting, any internal state is cleared. If no new metrics come in, the next interval will export nothing.
```

File: docs/sources/reference/components/otelcol/otelcol.processor.interval.md
Comment (clayton-cornell) [inline docs/sources/reference/components/otelcol/otelcol.processor.interval.md:53]:
```suggestion
Name          | Type       | Description                                                         | Default | Required
------------- | ---------- | ------------------------------------------------------------------- | ------- | --------
`interval`    | `duration` | The interval in the processor should export the aggregated metrics. | `"60s"` | no
```

File: docs/sources/reference/components/otelcol/otelcol.processor.interval.md
Comment (clayton-cornell) [inline docs/sources/reference/components/otelcol/otelcol.processor.interval.md:128]:
```suggestion

| Timestamp | Metric Name  | Aggregation Temporarility | Attributes        | Value |
```

Remove extra CR/LF

File: docs/sources/reference/components/otelcol/otelcol.processor.interval.md
Comment (clayton-cornell) [inline docs/sources/reference/components/otelcol/otelcol.processor.interval.md:143]:
```suggestion
The processor immediately passes the following metric to the next processor in the chain because it is a Delta metric.

| Timestamp | Metric Name  | Aggregation Temporarility | Attributes        | Value |
| --------- | ------------ | ------------------------- | ----------------- | ----: |
| 4         | other_metric | Delta                     | fruitType: orange |  77.4 |

```

File: docs/sources/reference/components/otelcol/otelcol.processor.interval.md
Comment (clayton-cornell) [inline docs/sources/reference/components/otelcol/otelcol.processor.interval.md:145]:
```suggestion
At the next `interval` (15s by default), the processor passed the following metrics to the next processor in the chain.
```

**Review by clayton-cornell — APPROVED** [review]
Some small suggestions and fixes. Otherwise the docs look OK.

**Comment by ArthurSens** [comment]
> Some small suggestions and fixes. Otherwise the docs look OK.

Thanks for taking a look!

**Pull request merged**
